odxtools/protstack.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+from xml.etree import ElementTree
+
+from .nameditemlist import NamedItemList
+from .comparamsubset import ComparamSubset
+from .element import IdentifiableElement
+from .exceptions import odxrequire
+from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId, OdxLinkRef
+from .utils import dataclass_fields_asdict
+
+if TYPE_CHECKING:
+    from .diaglayer import DiagLayer
+
+
+@dataclass
+class ProtStack(IdentifiableElement):
+    # mandatory in ODX 2.2, but non existent in ODX 2.0
+    pdu_protocol_type: str
+    physical_link_type: str
+    comparam_subset_refs: List[OdxLinkRef]
+
+    @staticmethod
+    def from_et(et_element: ElementTree.Element,
+                doc_frags: List[OdxDocFragment]) -> "ProtStack":
+
+        short_name = odxrequire(et_element.findtext("SHORT-NAME"))
+        doc_frags = [OdxDocFragment(short_name, str(et_element.tag))]
```

Comment:
this is only required/valid for hierarchy elements, i.e. the topmost elements contained by ODX files. Since protocol stacks are contained by `ComparamSpec`, this must be removed here:
```suggestion
```